src/context.rs

```diff
@@ -0,0 +1,132 @@
+//! Context related utilities
+//!
+//! This is the object that the worker inject into your application.
+//!
+//! You may want to check out [Context] to find the available variable
+//! and GitHub's documentation.
+//!
+//! To obtain the context, use the method [get_context] which hydrates the object
+//! with all the values from the environment variables
+use std::{env, fs, path::Path};
+
+use crate::error::ActionsError;
+use json::JsonValue;
+
+/// Context class injected by the action worker.
+///
+/// Find detailed documentation in
+/// [GitHub's documentation for Contexts](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)
+#[derive(Debug, Clone)]
+pub struct Context {
+    pub payload: JsonValue,
+    pub event_name: String,
+    pub sha: String,
+    pub ref_: String,
+    pub workflow: String,
+    pub action: String,
+    pub actor: String,
+    pub job: String,
+    pub run_attempt: u8,
+    pub run_number: u8,
+    pub run_id: u128,
+    pub api_url: String,
+    pub server_url: String,
+    pub graphql_url: String,
+    pub repo: Repo,
+}
+
+#[derive(Debug, Clone)]
+/// Container of two small values, owner and repo
+pub struct Repo {
+    pub owner: String,
+    pub repo: String,
+}
+
+/// Gets environment variables provided by GitHub and injects them into an object
+///
+/// Returns a [Context] object
+///
+/// Could return an [ActionsError] error type.
+/// ```rust,no_run
+/// use actions_github::context::get_context;
+/// let data = get_context().unwrap();
+/// println!("Event is {}", data.event_name);
+/// ```
+pub fn get_context() -> Result<Context, ActionsError> {
+    let mut payload: JsonValue = JsonValue::Null;
+
+    if let Ok(github_event_path) = env::var("GITHUB_EVENT_PATH") {
+        if Path::new(&github_event_path).exists() {
+            match fs::read_to_string(&github_event_path) {
+                Ok(content) => match json::parse(&content) {
+                    Ok(parsed_json) => payload = parsed_json,
+                    Err(err) => println!("Failed to parse JSON: {}", err),
+                },
+                Err(err) => println!("Failed to read file: {}", err),
+            }
+        } else {
+            println!("GITHUB_EVENT_PATH {} does not exist", github_event_path)
+        }
```

Comment:
Checking that file exists before reading is a discouraged practice. Even though it's fine in most cases, there aren't any guarantees that file will persist between check and the read. 

Instead, attempt reading file, and handle the "not found" error as you do here. 
Something like this:
```rust
            match fs::read_to_string(&github_event_path) {
                Ok(content) => match json::parse(&content) {
                    Ok(parsed_json) => payload = parsed_json,
                    Err(err) => println!("Failed to parse JSON: {}", err),
                },
                Err(err) if err.kind() === ErrorKind::NotFound => {
                    println!("GITHUB_EVENT_PATH {} does not exist", github_event_path),
                },
                Err(err) => println!("Failed to read file: {}", err),
            }
```
However, with two error handlers being basically identical, you can simply drop the `if Path::new(&github_event_path).exists() { ` clause.